src/base/reports/JsonReporter.hpp

```diff
@@ -4,28 +4,57 @@
 #pragma once
 
 #include "core/Config.hpp"
+#include "core/Time.hpp"
+#include "reports/ConsoleSink.hpp"
 #include "reports/IReporter.hpp"
 #include "reports/JsonPrinter.hpp"
 
 #include <string>
 
 namespace signal_estimator {
 
-class JsonReporter : public IReporter {
+template <typename Sink> class BasicJsonReporter final : public IReporter {
 public:
-    JsonReporter(const Config& config, const DevInfo& dev_info, JsonPrinter& printer);
-
-    JsonReporter(const JsonReporter&) = delete;
-    JsonReporter& operator=(const JsonReporter&) = delete;
-
-    void report(const LatencyReport& rep) override;
-    void report(const LossReport& rep) override;
-    void report(const IOJitterReport& rep) override;
-    void report(const IODelayReport& rep) override;
+    BasicJsonReporter(
+        const Config&, const DevInfo& dev_info, BasicJsonPrinter<Sink>& printer)
+        : dev_name_(dev_info.short_name)
+        , printer_(printer) {
+    }
+
+    void report(const LatencyReport& rep) override {
+        printer_.write("{\"type\": \"latency\", \"device\": \"{}\", \"timestamp\": {},"
+                       " \"sw_hw\": {}, \"hw\": {}, \"hw_avg\": {}}",
+            dev_name_, (unsigned long long)wallclock_timestamp_ns(), rep.sw_hw, rep.hw,
```

Comment:
The casts like `(unsigned long long)` are not needed anymore, the only reason for them was using printf.

Reply:
Good point, I hadn't looked into the type of the supplied timestamp

Reply:
Done. Also at other places that used c-style casts.